model_compression_toolkit/core/common/mixed_precision/sensitivity_evaluation.py

```diff
@@ -235,26 +225,27 @@ def _configure_bitwidths_model(self,
         # Thus, the last configurable layer must be included in the interest points for evaluating the metric,
         # otherwise, not all configurable nodes will be considered throughout the MP optimization search (since
         # they will not affect the metric value).
-        model_mp_layers_names = self.fw_impl.get_model_layers_names(model_mp)
         if node_idx is not None:  # configure specific layers in the mp model
             for node_idx_to_configure in node_idx:
-                node_name = self.get_quant_node_name(sorted_configurable_nodes_names[node_idx_to_configure])
-                if node_name in model_mp_layers_names:
-                    current_layer = self.fw_impl.get_model_layer_by_name(model_mp, node_name)
-                    self.set_layer_to_bitwidth(current_layer, mp_model_configuration[node_idx_to_configure])
-                else:
-                    raise Exception("The last configurable node is not included in the list of interest points for"  # pragma: no cover
-                                    "sensitivity evaluation metric for the mixed-precision search.")
-
+                self._configure_node_bitwidth(self.sorted_configurable_nodes_names,
+                                              mp_model_configuration, node_idx_to_configure)
         else:  # use the entire mp_model_configuration to configure the model
             for node_idx_to_configure, bitwidth_idx in enumerate(mp_model_configuration):
-                node_name = self.get_quant_node_name(sorted_configurable_nodes_names[node_idx_to_configure])
-                if node_name in model_mp_layers_names:
-                    current_layer = self.fw_impl.get_model_layer_by_name(model_mp, node_name)
-                    self.set_layer_to_bitwidth(current_layer, mp_model_configuration[node_idx_to_configure])
-                else:
-                    raise Exception("The last configurable node is not included in the list of interest points for"
-                                    "sensitivity evaluation metric for the mixed-precision search.")  # pragma: no cover
+                self._configure_node_bitwidth(self.sorted_configurable_nodes_names,
+                                              mp_model_configuration, node_idx_to_configure)
+
+    def _configure_node_bitwidth(self,
+                                 sorted_configurable_nodes_names,
+                                 mp_model_configuration,
+                                 node_idx_to_configure):
+        node_name = sorted_configurable_nodes_names[node_idx_to_configure]
+        layers_to_config = self.conf_node2layers.get(node_name, None)
+        if layers_to_config is None:
+            Logger.error(
+                f"Couldn't find matching layers in the MP model for node {node_name}.")  # pragma: no cover
+
+        for current_layer in layers_to_config:
```

Comment:
If I'm right, there should be a single layer to configure here, so why the list iteration?